File: descend_calc/descend_calc.py
def get_input(msg):
    while True:
        inp = input(msg)
        try:
            return int(inp)
        except ValueError:
            continue

def main():
    current_altitude = get_input("Current altitude: ")
    airport_altitude = get_input("Airport altitude: ")

    altitude_to_lose = ((current_altitude - airport_altitude) / 100) / 3 # remove last to digits

    current_speed = get_input("Current speed: ")
    avg_dsc_speed = get_input("Average descend speed: ")

    speed_knots = (current_speed - avg_dsc_speed) / 10

    wind_speed = get_input("Wind speed: ") / 10
    tail_wind = get_input("Tail or head wind [1 for tail | 0 for head]: ")

    descend_distance = altitude_to_lose + speed_knots

    if tail_wind == 1:
        descend_distance += wind_speed
    else:
        descend_distance -= wind_speed

    print("------------------------------")
    print(f"Altitude to lose nm: {altitude_to_lose}")
    print(f"Speed nm: {speed_knots}")
    print(f"Wind  nm: {wind_speed}")
    print(f"Total distance: {descend_distance}")

if __name__ == "__main__":
    main()

File: descend_calc/descend_calc.r
get_input <- function(msg)
{
    repeat
    {
        cat(msg)
        input <- trimws(readLines(con = "stdin", n = 1))
        if (grepl("^-?\\d+$", input))
        {
            return(as.integer(input))
        }
    }
}

main <- function()
{
    currentAltitude <- get_input("Current altitude: ")
    airportAltitude <- get_input("Airport altitude: ")

    altitudeToLose <- ((currentAltitude - airportAltitude) / 100.0) / 3.0

    currentSpeed <- get_input("Current speed: ")
    avgDscSpeed <- get_input("Average descend speed: ")

    speedKnots <- (currentSpeed - avgDscSpeed) / 10.0

    windSpeed <- get_input("Wind speed: ") / 10.0
    tailWind <- get_input("Tail or head wind [1 for tail | 0 for head]: ")

    descendDistance <- altitudeToLose + speedKnots + ifelse(tailWind == 1, windSpeed, -windSpeed)

    cat("------------------------------\n")
    cat(sprintf("Altitude to lose nm: %.4f\n", altitudeToLose))
    cat(sprintf("Speed nm: %.4f\n", speedKnots))
    cat(sprintf("Wind  nm: %.4f\n", windSpeed))
    cat(sprintf("Total distance: %.4f\n", descendDistance))
}

main()